pkg/plan/track_change.go

```diff
@@ -0,0 +1,178 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package plan
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/client/deployments"
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+	"github.com/elastic/cloud-sdk-go/pkg/util/slice"
+)
+
+// TrackChange iterates over a deployment's resources pending plans, sending
+// updates to the returned channel in the form of TrackResponse every frequency
+// configured in the parameter's TrackFrequencyConfig.
+// When all of deployment's resources pending plans have finished, the channel
+// is automatically closed by the goroutine that this function launches. It is
+// possible to iterate with a for loop and assume that the loop will exit after
+// all of the updates have been sent and the channel has been closed.
+// If a ResourceID and Kind are set instead of the DeploymentID, a reverse
+// lookup will be performed in order to find the DeploymentID and be able to
+// track the pending plan.
+func TrackChange(params TrackChangeParams) (<-chan TrackResponse, error) {
+	params.Config.fillDefaults()
+	if err := params.Validate(); err != nil {
+		return nil, err
+	}
+
+	deploymentID, err := getDeploymentID(params)
+	if err != nil {
+		return nil, err
+	}
+	params.DeploymentID = deploymentID
+
+	var out = make(chan TrackResponse)
+	go trackChange(params, out, time.NewTicker(params.Config.PollFrequency))
+
+	return out, nil
+}
+
+func trackChange(params TrackChangeParams, c chan<- TrackResponse, ticker *time.Ticker) {
```

Comment:
I don't have any preference, however since the legacy function name is Track, it might get confused and mixed up, thus why I chose `TrackChange` and `trackChange` as function names